### PR TITLE
workspace class for handling model serialization

### PR DESCRIPTION
**Pull request by ljj7975**
Please refer issue #3 for detail

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -153,29 +150,34 @@ def main(mode, config):
...
+
+
+        # load the best model
+        checkpoint = workspace.load_best_model()
 
         print("Training results")
-        print("\tbest_epoch: {0}".format(best_epoch))
-        print("\tbest_loss: {0}".format(best_loss))
-        print("\tbest_metric: {0}".format(best_metric))
+        print("\tbest_epoch: {0}".format(checkpoint["best_epoch"]))
```
Comment (daemon) [inline main.py]: {} is shorthand for {0}

File: utils/workspace.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            'optimizer_state_dict': self.optimizer.state_dict(),
+            'loss_fn': self.loss_fn,
+            'metric': self.metric
+            }
+
+        checkpoint.update(accessories)
+
+        torch.save(checkpoint, path)
+
+    def save_checkpoiunt(self, epoch, accessories):
```
Comment (daemon) [inline utils/workspace.py]: checkpoiunt misspelled

File: utils/workspace.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    def save_best_model(self, accessories):
+        path = self.__get_best_model_path()
+        self.__save(path, accessories)
+
+    def __load(self, path):
+        checkpoint = torch.load(path)
+
+        self.model.load_state_dict(checkpoint['model_state_dict'])
+        checkpoint.pop('model_state_dict')
```
Comment (daemon) [inline utils/workspace.py]: Does .pop return the value? If so, you can save a few lines.

File: utils/workspace.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        path = self.__get_checkpoint_path(epoch)
+
+        if not os.path.exists(path):
+            self.__save(path, accessories)
+
+    def save_best_model(self, accessories):
+        path = self.__get_best_model_path()
+        self.__save(path, accessories)
+
+    def __load(self, path):
```
Comment (daemon) [inline utils/workspace.py]: Isn't double underscore reserved for Python routines?

**Review by daemon — APPROVED** [review]

**Comment by ljj7975** [comment]
closes #3

**Pull request merged**